// keystone/admin/routes/views/register.js
var keystone = require('../../../');
var session = require('../../../lib/session');
var url = require('url');
var async = require('async');
var User = keystone.list('User');

exports = module.exports = function(req, res) {

	function renderView() {
		keystone.render(req, res, 'register', {
			submitted: req.body,
			from: req.query.from,
			logo: keystone.get('register logo')
		});
	}

	function createUser (admin, done) {
		var user = new User.model(admin);

		user.isAdmin = false;
		user.save(function (err) {
			if (err) {
				console.error('Error adding user ' + admin.email + ' to the database:');
				console.error(err);
			} else {
				console.log('Added user ' + admin.email + ' to the database.');
			}
			// done(err);
		});
	}

	// If a form was submitted, process the login attempt
	if (req.method === 'POST') {

		if (!keystone.security.csrf.validate(req)) {
			req.flash('error', 'There was an error with your request, please try again.');
			return renderView();
		}

		if (!req.body.fname || !req.body.lname || !req.body.email || !req.body.password) {
			req.flash('error', 'Please enter your name, email address and password.');
			return renderView();
		}

		var user = {
			email: req.body.email,
			password: req.body.password,
			name: {
				first: req.body.fname,
				last: req.body.lname
			},
			location: req.body.location,
			post: req.body.post,
			company: req.body.company,
			qualification: req.body.qualification,
			university: req.body.university,
			party: req.body.party
		};

		createUser(user);

		res.redirect('/admin/signin');
		// var onSuccess = function (user) {

			
		// };

		// var onFail = function (err) {
		// 	var message = (err && err.message) ? err.message : 'Sorry, that email and password combo are not valid.';
		// 	req.flash('error', message );
		// 	renderView();
		// };

		// session.signin(req.body, req, res, onSuccess, onFail);

	} else {
		renderView();
	}

};
